docs/appendices/release-notes/unreleased.rst

```diff
@@ -248,6 +248,10 @@ Others
 Fixes
 =====
 
+- Allow user to control the maximum number of threads on a single shard
```

Comment:
Did you intend for this to be backported to 4.0? Usually this kind of change only goes into a feature release (4.1) - in which case we'd list it somewhere under `Changes`.

Reply:
Sorry, I will move it under `Changes`.